spring-kafka/src/main/java/org/springframework/kafka/config/RetryTopicComponentFactory.java

```diff
@@ -0,0 +1,172 @@
+/*
+ * Copyright 2022 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.kafka.config;
+
+import java.time.Clock;
+
+import org.springframework.beans.factory.BeanFactory;
+import org.springframework.kafka.listener.DeadLetterPublishingRecoverer;
+import org.springframework.kafka.listener.KafkaConsumerBackoffManager;
+import org.springframework.kafka.listener.adapter.KafkaBackoffAwareMessageListenerAdapter;
+import org.springframework.kafka.retrytopic.DeadLetterPublishingRecovererFactory;
+import org.springframework.kafka.retrytopic.DefaultDestinationTopicProcessor;
+import org.springframework.kafka.retrytopic.DefaultDestinationTopicResolver;
+import org.springframework.kafka.retrytopic.DestinationTopic;
+import org.springframework.kafka.retrytopic.DestinationTopicProcessor;
+import org.springframework.kafka.retrytopic.DestinationTopicResolver;
+import org.springframework.kafka.retrytopic.ListenerContainerFactoryConfigurer;
+import org.springframework.kafka.retrytopic.ListenerContainerFactoryResolver;
+import org.springframework.kafka.retrytopic.RetryTopicConfiguration;
+import org.springframework.kafka.retrytopic.RetryTopicConfigurer;
+import org.springframework.kafka.retrytopic.RetryTopicNamesProviderFactory;
+import org.springframework.kafka.retrytopic.SuffixingRetryTopicNamesProviderFactory;
+
+/**
+ * Provides the component instances that will be used with
+ * {@link RetryTopicConfigurationSupport}. Override any of the methods to provide
+ * a different implementation or subclass, then override the
+ * {@link RetryTopicConfigurationSupport#createComponentFactory()} method
+ * to return this factory's subclass.
+ *
+ * @author Tomaz Fernandes
+ * @since 2.9
+ */
+public class RetryTopicComponentFactory {
+
+	private final Clock internalRetryTopicClock = createInternalRetryTopicClock();
+
+	/**
+	 * Creates the {@link RetryTopicConfigurer} that will serve as an entry point
```

Comment:
The method Javadoc must be imperative: https://github.com/spring-projects/spring-framework/wiki/Code-Style#javadoc-formatting